dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/jsonpatch/JsonPatchManager.java

```diff
@@ -73,7 +78,7 @@ public JsonPatchManager(
      * @param object Jackson Object to apply the patch to.
      * @return New instance of the object with the patch applied.
      */
-    @Transactional
+    @Transactional( readOnly = true )
```

Comment:
Now the big question is: can this cause creation of new periods?

Reply:
in this patchManager, we only get the data and convert to jsonNode, then patch the jsonNode object. 
No saving data to database, so the Periods will just be copied to the jsonNode, no matter if its existed or not.
The saving/updating will happen  in the importService, where the write transaction should be handled.